### ISSUE TITLE
Used softmax activation at the policy head.

### ISSUE THREAD

**Issue by Gruhit13** [BentouAI/AlphaZero-Chain-Reaction#2]
Hello sir, I have seen your tutorial on Medium for the same and it is fantastic. Although, the code you provided here has a minor issue in that you are using Softmax activation on the final linear layer for Policy Head for calculating loss you have used `nn.CrossEntropyLoss()` that accepts raw logits as input and probability distribution for target and it internally performs the softmax for the same. Thus passed already softmax result might give you improper loss value and hence can result in wrong training of the model.
